src/common/grouping.c

```diff
@@ -181,11 +180,10 @@ void dt_grouping_add_grouped_images(GList **images)
         sqlite3_finalize(stmt);
       }
     }
-    imgs = g_list_next(imgs);
   }
 
   if(gimgs)
-    imgs = g_list_concat(*images, g_list_reverse(gimgs));
```

Comment:
This is not equivalent ?

Reply:
It is with the current code, but the modified version is more robust.  It won't break if the code is ever modified so that it could invoke that line when *images is NULL, and it won't break the compile if unused-variable warnings are enabled.

Reply:
Yes I see. Thanks.